packages/pytest-simcore/src/pytest_simcore/helpers/s3.py

```diff
@@ -77,36 +79,29 @@ async def upload_file_to_presigned_link(
 ) -> list[UploadedPart]:
     file_size = file.stat().st_size
 
-    start = perf_counter()
-    print(f"--> uploading {file=}")
-    async with ClientSession() as session:
-        file_chunk_size = int(file_upload_link.chunk_size)
-        num_urls = len(file_upload_link.urls)
-        last_chunk_size = file_size - file_chunk_size * (num_urls - 1)
-        upload_tasks = []
-        for index, upload_url in enumerate(file_upload_link.urls):
-            this_file_chunk_size = (
-                file_chunk_size if (index + 1) < num_urls else last_chunk_size
-            )
-            upload_tasks.append(
-                upload_file_part(
-                    session,
-                    file,
-                    index,
-                    index * file_chunk_size,
-                    this_file_chunk_size,
-                    num_urls,
-                    upload_url,
+    with log_context(logging.INFO, msg=f"uploading {file} via {file_upload_link=}"):
+        async with ClientSession() as session:
+            file_chunk_size = int(file_upload_link.chunk_size)
+            num_urls = len(file_upload_link.urls)
+            last_chunk_size = file_size - file_chunk_size * (num_urls - 1)
+            upload_tasks = []
+            for index, upload_url in enumerate(file_upload_link.urls):
+                this_file_chunk_size = (
+                    file_chunk_size if (index + 1) < num_urls else last_chunk_size
                 )
-            )
-        results = await logged_gather(*upload_tasks, max_concurrency=0)
-    part_to_etag = [
-        UploadedPart(number=index + 1, e_tag=e_tag) for index, e_tag in results
-    ]
-    print(
-        f"--> upload of {file=} of {file_size=} completed in {perf_counter() - start}"
-    )
-    return part_to_etag
+                upload_tasks.append(
+                    upload_file_part(
+                        session,
+                        file,
+                        index,
+                        index * file_chunk_size,
+                        this_file_chunk_size,
+                        num_urls,
+                        upload_url,
+                    )
+                )
+            results = await logged_gather(*upload_tasks, max_concurrency=0)
```

Comment:
I had a crazy thought the other day: Since S3 forces us to download the content and upload it again. Wouldn't it makes sense to forward this to the user. Just like we upload directly from the user side instead of passing content through our servers, why don't we implement the copying endpoint on the "client side" (meaning either the frontend or the dynamic sidecar I guess). That way we would spare `storage` from all the heavy lifting

Reply:
so, concerning copying it is a bit more complex.
- if we copy below 5GB, the copies appear to happen in place, provided it is the same region, etc etc..
- I am still not 100% sure of when doing multipart copy whether we can also leverage it for copying parts, but I have already spent quite some time and need to also look at the EBS stuff.
- I would like to see if we get some improvements with this now.